cli/src/semgrep/rule.py

```diff
@@ -261,6 +262,24 @@ def product(self) -> RuleProduct:
             else RuleProduct.sast
         )
 
+    @property
+    def scan_source(self) -> RuleScanSource:
+        src: str = self.metadata.get("semgrep.dev", {}).get("src", "")
+        if src == "unchanged":
+            return RuleScanSource.unchanged
+        elif src == "new-version":
+            return RuleScanSource.new_version
+        elif src == "new-rule":
+            return RuleScanSource.new_rule
+        elif src == "previous-scan":
+            return RuleScanSource.previous_scan
+        else:
+            return RuleScanSource.unannotated
+
+    @property
+    def is_prev_scan(self) -> bool:
```

Comment:
it isn't immediately clear what is_prev_scan means.

Does this mean it existed in a previous scan and still exists now?

Maybe it's better to call this something like `for_transient_scan` or something more specific than previous scan.

Reply:
did this!